lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -144,6 +166,24 @@ LogicalResult mlir::substrait::DecimalType::verify(
   return success();
 }
 
+LogicalResult mlir::substrait::ListType::verify(
+    llvm::function_ref<mlir::InFlightDiagnostic()> emitError, mlir::Type type) {
+  // ListType must be parameterized with an atomic type.
+  if (!mlir::isa<mlir::IntegerType, mlir::Float32Type, mlir::Float64Type,
+                 substrait::StringType, substrait::BinaryType,
+                 substrait::TimestampType, substrait::TimestampTzType,
+                 substrait::DateType, substrait::TimeType,
+                 substrait::IntervalYearMonthType,
+                 substrait::IntervalDaySecondType, substrait::UUIDType,
+                 substrait::FixedCharType, substrait::VarCharType,
+                 substrait::FixedBinaryType, substrait::DecimalType>(type))
+    return emitError()
+           << "ListType must be parameterized with an atomic type, but got: "
```

Comment:
Nit: single-quoted class name.
```suggestion
           << "'ListType' must be parameterized with an atomic type, but got: "
```